DigitalLearningSolutions.Data/Services/DelegateApprovalsService.cs

```diff
@@ -30,5 +44,69 @@ public List<(DelegateUser delegateUser, List<CustomPromptWithAnswer> prompts)> G
 
             return usersWithPrompts;
         }
+
+        public void ApproveDelegate(int delegateId)
+        {
+            var delegateUser = userDataService.GetDelegateUserById(delegateId);
+
+            if (delegateUser == null)
+            {
+                throw new UserAccountNotFoundException($"Delegate user id {delegateId} not found");
+            }
+            if (delegateUser.Approved)
+            {
+                logger.LogWarning($"Delegate user id {delegateId} already approved.");
+            }
+            else
+            {
+                userDataService.ApproveDelegateUsers(new[] { delegateId });
+                SendApprovalEmail(delegateUser);
+            }
+        }
+
+        public void ApproveAllUnapprovedDelegatesForCentre(int centreId)
+        {
+            var delegateUsers = userDataService.GetUnapprovedDelegateUsersByCentreId(centreId);
+
+            userDataService.ApproveDelegateUsers(delegateUsers.Select(d => d.Id));
+            foreach (var delegateUser in delegateUsers)
+            {
+                SendApprovalEmail(delegateUser);
+            }

```

Comment:
I wonder if we could have SendEmails method in the EmailService so we don't need to connect-disconnect many times (as I suspect that's the bottleneck in terms of performance)